haystack/nodes/prompt/prompt_node.py

```diff
@@ -326,6 +332,10 @@ def __init__(
             use_auth_token=self.use_auth_token,
             model_kwargs=model_input_kwargs,
         )
+        # This is how the default max_length is determined for Text2TextGenerationPipeline shown here
+        # https://huggingface.co/transformers/v4.6.0/_modules/transformers/pipelines/text2text_generation.html
+        # max_length must be set otherwise HFLocalInvocationLayer._ensure_token_limit will fail.
+        self.max_length = max_length or self.pipe.model.config.max_length
```

Comment:
I agree this is confusing, but `max_length` is the language used by HuggingFace transformers which is why I think we went with it here. And to be clear the OpenAI API names this variable max_tokens, but we've mapped it max_length for consistency. 

I think if we made this change we should make sure to rename `max_length` everywhere including the variable names used to initialize the `HFLocalInvocationLayer` and `OpenAIInvocationLayer` classes otherwise we risk more confusion I think. I would recommend opening this as a separate PR and potentially labeling it breaking since it might affect existing code.